src/ExponentialBackoff.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace ipl\Stdlib;
+
+use Exception;
+use LogicException;
+
+class ExponentialBackoff
+{
+    /** @var int The minimum wait time for each retry in ms */
+    protected $min;
+
+    /** @var int The maximum wait time for each retry in ms */
+    protected $max;
+
+    /** @var int Number of retries to be performed before giving up */
+    protected $retries;
+
+    /** @var ?int The previous used retry wait time */
+    protected $previousWaitTime;
```

Comment:
I would not maintain state and just do `$min << $attempt` as in our go code.

Reply:
He didn't do that on my request.

Reply:
Please justify.

Reply:
Unnecessarily complex and difficult to understand for no apparent advantage

Reply:
```php

Reply:
> Unnecessarily complex and difficult to understand for no apparent advantage

You can't be serious. If this is too complex (`2 ** $attempt * $min`), I wonder how we can build software.

The obvious advantage is that no state is introduced without reason, which is then also **changed in a getter**. 
The ipl should be an example of good coding practice. The current implementation is not.

The seemingly not so obvious advantage is that this does not fall apart when running asynchronously. Sure, PHP isn't ready to support everything concurrently yet, but PHP isn't the only programming language and there are fibers, async I/O (ReactPHP and the like) that already make this possible. When state like this one is introduced, concurrency should also be thought of. Even in PHP.

Also, `getWaitTime()` claims to return a proper wait time in the first line and then explains in the description that it actually does not. Calling `getWaitTime(n)` yields wrong results without having called `getWaitTime(0..n-1)` first.
Plus, when fixed, the actual backoff implementation would be properly testable.